benchmark/benchmarkTests.go

```diff
@@ -0,0 +1,338 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+        "fmt"
+	"testing"
+
+	"github.com/awslabs/soci-snapshotter/benchmark/framework"
+	"github.com/containerd/containerd"
+)
+
+func BenchmarkPullImageFromECR(b *testing.B, imageRef string) {
+	containerdProcess, err := getContainerdProcess() 
+	if err != nil {
+                fmt.Printf("Error Starting Containerd: %v\n", err)
+		b.Fatal(err)
+	}
+	defer containerdProcess.StopProcess()
+	b.ResetTimer()
+	_, err = containerdProcess.PullImageFromECR(imageRef, platform, awsSecretFile)
```

Comment:
`PullImageFromRegistry`? Ideally, we'd like to use other registries, not only ECR.

Reply:
This specifically uses the resolver necessary for ECR, in containerd_utils there is a function that does not use, but any registry that does not require login
```
func (proc *ContainerdProcess) PullImage(
	imageRef string,
	platform string) (containerd.Image, error)
```